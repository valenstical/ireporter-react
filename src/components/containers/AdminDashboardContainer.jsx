import React, { Component } from 'react';
import { createStructuredSelector } from 'reselect';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { getUser, getReports, getSummary } from '../../selectors';
import fetchReports from '../../actions/getReportAction';
import AdminDashboard from '../views/AdminDashboard';

/**
 * Container component for the AdminDashboard view
 * @export
 * @class AdminDashboardContainer
 * @extends {Component}
 */
class AdminDashboardContainer extends Component {
  componentDidMount() {
    this.handleGetReports();
  }

  handleGetReports = () => {
    const { fetchReports: processFetchReports } = this.props;
    processFetchReports();
  }

  render() {
    const { reports, summary, user } = this.props;
    if (!user.isLoggedIn || !user.isAdmin) return <Redirect to="/login" />;
    return (
      reports.data
        ? (
          <AdminDashboard
          reports={reports}
          summary={summary}
          active={reports.active}
          handleGetReports={this.handleGetReports}
         />
        )
        : null
    );
  }
}

AdminDashboardContainer.propTypes = {
  user: PropTypes.object.isRequired,
  reports: PropTypes.object.isRequired,
  summary: PropTypes.object.isRequired,
  fetchReports: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector(
  {
    user: getUser,
    reports: getReports,
    summary: getSummary
  }
);

const mapDispatchToProps = dispatch => ({
  fetchReports: () => dispatch(fetchReports('incidents/admin', true, 'incidents')),
});

export default connect(mapStateToProps, mapDispatchToProps)(AdminDashboardContainer);
