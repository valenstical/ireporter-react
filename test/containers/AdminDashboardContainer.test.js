import React from 'react';
import setup from '../setup';
import AdminDashboardContainer from '../../src/components/containers/AdminDashboardContainer';


describe('<AdminDashboardContainer>', () => {
  it('should render without crashing', () => {
    const wrapper = setup(<AdminDashboardContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
