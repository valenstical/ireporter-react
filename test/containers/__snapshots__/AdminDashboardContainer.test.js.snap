// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminDashboardContainer> should render without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "REPLACE",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/login",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(AdminDashboardContainer)>
        <AdminDashboardContainer
          fetchReports={[Function]}
          reports={
            Object {
              "data": Array [],
              "isBusy": false,
              "success": false,
            }
          }
          summary={
            Object {
              "investigating": 0,
              "pending": 0,
              "rejected": 0,
              "resolved": 0,
            }
          }
          user={
            Object {
              "isBusy": false,
              "message": Array [],
              "success": false,
            }
          }
        >
          <Redirect
            push={false}
            to="/login"
          />
        </AdminDashboardContainer>
      </Connect(AdminDashboardContainer)>
    </Router>
  </BrowserRouter>
</Provider>
`;
